warehouse/slave.go

```diff
@@ -352,9 +352,19 @@ func processStagingFile(job PayloadT, workerIndex int) (loadFileUploadOutputs []
 	jobRun.setStagingFileDownloadPath(workerIndex)
 
 	// This creates the file, so on successful creation remove it
-	err = jobRun.downloadStagingFile()
+	err = jobRun.downloadStagingFile(jobRun.job.DestinationConfig)
 	if err != nil {
-		return loadFileUploadOutputs, err
+		// If error occurs with the current config
+		// We retry with the revision config if it is present
+		if jobRun.job.DestinationRevisionConfig != nil {
+			pkgLogger.Infof("[WH]: Starting downloading staging file with revision config for StagingFileID: %d, StagingFileLocation: %s, whIdentifier: %s", job.StagingFileID, job.StagingFileLocation, jobRun.whIdentifier)
+			err = jobRun.downloadStagingFile(jobRun.job.DestinationRevisionConfig)
+			if err != nil {
+				return loadFileUploadOutputs, err
```

Comment:
Send some stats around this if both of the case will fail. Also log both the revisionId and current config.

Reply:
From @abhimanyubabbar

Reply:
Done.
Added this stat worker_processing_download_staging_file_failed.